### PR TITLE
Give jobs a second chance

### PR DESCRIPTION
**Pull request by Devidxyz**
Promlem & solution here: https://discord.com/channels/697041998728659035/1100897398454222982/1176954485298450595

Tl;dr 
normal flow:
- worker picks up a job
- worker puts lock key in redis
- monitor check if all jobs in processing queue has a lock key in redis => it has a lock no problem

inconsistency:
- worker picks up a job
- monitor check if all jobs in processing queue has a lock key in redis => it will mark the job as failed
- worker puts lock key in redis

solution:
- worker picks up a job
- monitor check if all jobs in processing queue has a lock key in redis => monitor remembers that failed for the first time
- worker puts lock key in redis
- monitor check if all jobs in processing queue has a lock key in redis => it has a lock, no problem

### CODE REVIEW

**Review by ejayxyz — APPROVED** [review]

**Review by PAStheLoD — APPROVED** [review]
🪄
